src/box/replication.cc

```diff
@@ -314,8 +313,9 @@ replica_delete(struct replica *replica)
 	assert(replica_is_orphan(replica));
 	if (replica->relay != NULL)
 		relay_delete(replica->relay);
-	if (replica->gc != NULL)
-		gc_consumer_unregister(replica->gc);
+	/* Drop consumer only if another replica does not own it. */
+	if (replica_by_uuid(&replica->uuid) == NULL)
```

Comment:
This check is essential. Dropped a comment here:
```c
/*
 * Drop consumer only if another replica does not own it.
 * Another replica can own it, for example, when we created a new
 * replica object on `replicaset_update`, and now we drop it because
 * we can reuse the old one.
 */
```